scheduler/src/cook/mesos/util.clj

```diff
@@ -26,6 +26,44 @@
             [plumbing.core :as pc :refer (map-vals map-keys)])
   (:import [java.util Date]))
 
+(defn lookup-cache-generic!
+      "Generic cache. Caches under a key (extracted from the item with extract-key-fn. Uses miss-fn to fill
+      any misses. Caches only postive hits where both functions return non-nil"
+      [cache extract-key-fn miss-fn item]
+      (let [key (extract-key-fn item)]
+           (if key
+             (cache/lookup (swap! cache
+                                  #(if (cache/has? % key)
+                                     (cache/hit % key)
+                                     (let [result (miss-fn item)]
+                                          ; Only cache non-nil
+                                          (if result
+                                            (cache/miss % key result)
+                                            %))))
+                           key)
+             (miss-fn item))))
+
+(defn lookup-cache-datomic-entity!
+      "Specialized function for caching where datomic entities are the key. Extracts :db/id so that
+      we don't keep the entity alive in the cache"
+      [cache miss-fn entity]
+      (lookup-cache-generic! cache :db/id miss-fn entity))
+
+; Fake. Are actually initialized in flush-caches
+(defonce job-resources-cache (atom nil))
```

Comment:
Can we initialize the caches here? It doesn't seem like it would be any faster to `def` them with nil and call `flush-caches` with 1,000,000 while loading the file.